#! /bin/sh /usr/share/dpatch/dpatch-run
## 93_console-1.5.0.dpatch by Thomas Günther <tom@toms-cafe.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Changes for VDR >= 1.5.0 (applicable to VDR >= 1.4.5).

@DPATCH@
--- console-0.6.0/menu.c
+++ console-0.6.0/menu.c
@@ -1193,7 +1193,7 @@
 
     switch (BASICKEY(Key)) {
 
-    case kKbd:{     uint64 buf = cKbdRemote::MapFuncToCode(KEYKBD(Key));
+    case kKbd:{     uint64_t buf = cKbdRemote::MapFuncToCode(KEYKBD(Key));
                     WriteToConsole(buf);
 
                     // On every key press, we show the cursor immediately
@@ -1425,7 +1425,7 @@
 
 
 
-void cMenuConsole::WriteToConsole(const uint64& code) {
+void cMenuConsole::WriteToConsole(const uint64_t& code) {
 
   unsigned char* data = (unsigned char*)(void*)&code;
 
--- console-0.6.0/menu.h
+++ console-0.6.0/menu.h
@@ -102,7 +102,7 @@
 private:
   void CaptureKeyboard();
   void ReleaseKeyboard();
-  void WriteToConsole(const uint64& code);
+  void WriteToConsole(const uint64_t& code);
   bool TerminateConsole();
 
   void AddSubMenu(cOsdObject* pSubMenu);
